src/nomad_simulations/physical_property.py

```diff
@@ -0,0 +1,231 @@
+            - scalars (tensor rank 0) have `rank=[]` (`len(rank) = 0`),
+            - vectors (tensor rank 1) have `rank=[a]` (`len(rank) = 1`),
+            - matrices (tensor rank 2), have `rank=[a, b]` (`len(rank) = 2`),
+            - etc.
+        """,
+    )
+
+    variables = SubSection(sub_section=Variables.m_def, repeats=True)
+
+    value: Quantity = _placeholder_quantity
+
+    entity_ref = Quantity(
+        type=Entity,
+        description="""
+        Reference to the entity that the physical property refers to.
+        """,
+    )
+
+    physical_property_ref = Quantity(
+        type=Reference(SectionProxy('PhysicalProperty')),
+        description="""
+        Reference to the `PhysicalProperty` section from which the physical property was derived. If `physical_property_ref`
+        is populated, the quantity `is_derived` is set to True via normalization.
+        """,
+    )
+
+    is_derived = Quantity(
+        type=bool,
+        default=False,
+        description="""
+        Flag indicating whether the physical property is derived from other physical properties. We make
+        the distinction between directly parsed and derived physical properties:
+            - Directly parsed: the physical property is directly parsed from the simulation output files.
+            - Derived: the physical property is derived from other physical properties. No extra numerical settings
+                are required to calculate the physical property.
+        """,
+    )
+
+    is_scf_converged = Quantity(
+        type=bool,
+        description="""
+        Flag indicating whether the physical property is converged or not after a SCF process. This quantity is connected
+        with `SelfConsistency` defined in the `numerical_settings.py` module.
+        """,
+    )
+
+    self_consistency_ref = Quantity(
+        type=SelfConsistency,
+        description="""
+        Reference to the `SelfConsistency` section that defines the numerical settings to converge the
+        physical property.
+        """,
+    )
+
+    @property
+    def variables_shape(self) -> Optional[list]:
+        """
+        Shape of the variables over which the physical property varies. This is extracted from
+        `Variables.n_grid_points` and appended in a list.
+
+        Example, a physical property which varies with `Temperature` and `ElectricField` will
+        return `variables_shape = [n_temperatures, n_electric_fields]`.
+
+        Returns:
+            (list): The shape of the variables over which the physical property varies.
+        """
+        if self.variables is not None:
+            return [v.get_n_grid_points(v.grid_points, logger) for v in self.variables]
+        return []
+
+    @property
+    def full_shape(self) -> list:
+        """
+        Full shape of the physical property. This quantity is calculated as:
+            `full_shape = variables_shape + shape`
```

Comment:
Be careful, cause this sounds as if you are talking about numbers.
The proper term for lists is _concatenation_. Python calls it `__add__()` so they can use the built-in symbol.

Reply:
Cool, thanks a lot, I could never remember _concatenation_, I was always using the concept of _adding_ two lists 😄  I will use it in the description, as I think it is easier to understand.